src/game.cpp

```diff
@@ -1183,7 +1183,12 @@ ReturnValue Game::internalMoveItem(Cylinder* fromCylinder, Cylinder* toCylinder,
 
 	//add item
 	if (moveItem /*m - n > 0*/) {
-		toCylinder->addThing(index, moveItem);
+		if (fromCylinder == toCylinder) {
+			toCylinder->addThing(index, moveItem);
+			
+		} else {
+			internalAddItem(toCylinder, moveItem, INDEX_WHEREEVER, flags);
```

Comment:
Just FYI this is somehow bugged. When you move item from bp to eq slot and your cap is not enough to carry the item you are moving then item will dissappear.